src/main/java/duke/tasks/Deadline.java

```diff
@@ -0,0 +1,54 @@
+package duke.tasks;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * A class that represents the a task that has a deadline.
+ */
+public class Deadline extends Task {
+    private LocalDateTime by;
```

Comment:
It might be a better idea to name the variable for LocalDateTime, something more clear, instead of just calling it "by".
For example, "toBeCompletedBy".